Add an incredibly lame pun

throwawayuseless opened the pull request:
## About The Pull Request
Grenadine now causes dizzyness and minor stamina damage in revolutionaries. Because pomegranate stops **FREE RADICALS**.
## Why It's Good For The Game
Obscure and funny drink effects are obscure and funny. Plus, changelings and clown cars also have obscure reagents that can counter them if you're truly desperate. Also included is an **additional joke** in _this very PR!_ Q: "Why do marxists love herbal infusions?" A: "Because all proper tea is theft!"
## Changelog
:cl:
add: Revolutionaries are now weakened by grenadine. Because the pomegranate antioxidants stop free radicals.
/:cl:

throwawayuseless committed:
Add incredibly lame pun.

hehe

throwawayuseless committed:
fix bad template pasting

MrMelbert commented on code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm:
```diff
@@ -994,6 +994,17 @@
 
+/datum/reagent/consumable/grenadine/on_mob_add(mob/living/drinker)
+	. = ..()
+	if(drinker.mind && drinker.mind.has_antag_datum(/datum/antagonist/rev))
```
```suggestion
	if(IS_REVOLUTIONARY(drinker))
```

MrMelbert commented on code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm:
```diff
@@ -994,6 +994,17 @@
+		
+/datum/reagent/consumable/grenadine/on_mob_life(mob/living/carbon/drinker, seconds_per_tick, times_fired)
+	. = ..()
+	if(drinker.mind && drinker.mind.has_antag_datum(/datum/antagonist/rev))
```
```suggestion
	if(IS_REVOLUTIONARY(drinker))
```

MrMelbert commented on code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm:
```diff
@@ -994,6 +994,17 @@
 	taste_description = "sweet pomegranates"
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
+/datum/reagent/consumable/grenadine/on_mob_add(mob/living/drinker)
```
```suggestion
/datum/reagent/consumable/grenadine/on_mob_metabolize(mob/living/drinker)
```

MrMelbert commented on code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm:
```diff
@@ -994,6 +994,17 @@
+	. = ..()
+	if(drinker.mind && drinker.mind.has_antag_datum(/datum/antagonist/rev))
+		drinker.set_dizzy_if_lower(10 SECONDS * REM * seconds_per_tick)
+		drinker.adjustStaminaLoss(12, required_biotype = affected_biotype) //The pomegranate stops free radicals! Har har.
```
Maybe cap the stamina damage before stamcrit or people will legit use this in syringe guns on rev rounds, lol

throwawayuseless replied:
I've changed it so that it will only ever push you to the edge of stamcrit, no actual stamcritting from just the grenadine. You will get stamcrit if you take a good whack with a stamina weapon though.

MrMelbert reviewed (changes requested):
Awesome

throwawayuseless committed:
Update code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

throwawayuseless committed:
Use better rev check

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

throwawayuseless committed:
Update code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm

Better rev check

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

throwawayuseless committed:
Merge branch 'tgstation:master' into lamepun

throwawayuseless committed:
Cannot stamcrit.

Grenadine will now push revs to the edge of stamina crit, without ever actually stamcritting them. At least one good whack with a stamina based weapon or other stamina toxin is now required to actually immobilize them.

throwawayuseless committed:
Whoops stupid number wont work as intended

throwawayuseless committed:
Merge branch 'tgstation:master' into lamepun

throwawayuseless committed:
DAMN YOU PYTHON MUSCLE MEMORY

silly linter

throwawayuseless commented:
stalebot moment *cry

throwawayuseless commented:
Woe: stalebot be upon me

JohnFulpWillard commented:
$2
